src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/CodeDomLocalizationProvider.LanguageExtenders.cs

```diff
@@ -0,0 +1,267 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+using System.Windows.Forms;
+using System.Windows.Forms.Design;
+
+namespace System.ComponentModel.Design.Serialization
+{
+    public sealed partial class CodeDomLocalizationProvider
+    {
+        /// <summary>
+        ///  The design time language and localizable properties.
+        /// </summary>
+        [ProvideProperty("Language", typeof(IComponent))]
+        [ProvideProperty("LoadLanguage", typeof(IComponent))]
+        [ProvideProperty("Localizable", typeof(IComponent))]
+        internal class LanguageExtenders : IExtenderProvider
+        {
+            private readonly IServiceProvider _serviceProvider;
+            private readonly IDesignerHost _host;
+            private IComponent _lastRoot;
+            private readonly TypeConverter.StandardValuesCollection _supportedCultures;
+            private bool _localizable;
+            private CultureInfo _language;
+            private CultureInfo _loadLanguage;
+            private CultureInfo _defaultLanguage;
+
+            public LanguageExtenders(IServiceProvider serviceProvider, CultureInfo[] supportedCultures)
+            {
+                _serviceProvider = serviceProvider;
+                _host = serviceProvider.GetService(typeof(IDesignerHost)) as IDesignerHost;
```

Comment:
Lets use extension methods whereever possible.
```suggestion
                _host = serviceProvider.GetService<IDesignerHost>()
```